libs/server/AOF/AofProcessor.cs

```diff
@@ -158,44 +167,64 @@ internal unsafe void ProcessAofRecord(IMemoryOwner<byte> entry, int length, bool
         {
             fixed (byte* ptr = entry.Memory.Span)
             {
-                ProcessAofRecordInternal(ptr, length, asReplica);
+                if (replayFromLegacyAof && IsLegacyFormat(ptr))
```

Comment:
Should we return an error if replayFromLegacyAof is set but IsLegacyFormat is false?

Reply:
this is on purpose, because this handles a server that was started in "replayFromLegacyMode" but then starts reading new aof format logs.
You dont need to worry about the log being new and having server in this backwards compat mode.